src/it/scala/com/northpole/santas/NaughtyNiceReportSpec.scala

```diff
@@ -1,174 +1,169 @@
+    for {
+      _ <- fixture().behaviourReportProducer
+        .produceOne_(
+          ProducerRecord(
+            BehaviourReportTopic,
+            MegachusFullName,
+            BehaviourReport(ExpectedScore)
+          )
         )
-      val addressUpdateProducer =
-        ProducerSettings[IO, FullName, Address].withBootstrapServers(
-          kafkaContainer.bootstrapServers
+        .flatten
+      _ <- fixture().addressUpdateProducer
+        .produceOne_(
+          ProducerRecord(
+            AddressUpdateReportTopic,
+            MegachusFullName,
+            MegachusAddress
+          )
+        )
+        .flatten
+      // Assert
+      megachusHouseInhabitants <- eventually(
+        getHouseInhabitants(fixture().applicationServer.baseUri, MegachusAddress)
+      )
+      _ <- eventually(
+        assertIO(
+          getConsignment(fixture().applicationServer.baseUri, megachusHouseInhabitants.head),
+          ChristmasConsignment(
+            MegachusFullName,
+            MegachusAddress.some,
+            ChristmasPresent.Nice(uri"nice_present", ExpectedScore)
+          )
         )
-// Action
-      (
-        KafkaProducer.resource(behaviourReportProducer),
-        KafkaProducer.resource(addressUpdateProducer),
-        SantasServer.resource(config)
-      ).tupled
-        .use { case (behaviourReportProducer, addressUpdateProducer, server) =>
-          val ExpectedScore = 7
-          for {
-            _ <- behaviourReportProducer
-              .produceOne_(
-                ProducerRecord(
-                  BehaviourReportTopic,
-                  MegachusFullName,
-                  BehaviourReport(ExpectedScore)
-                )
-              )
-              .flatten
-            _ <- addressUpdateProducer
-              .produceOne_(
-                ProducerRecord(
-                  AddressUpdateReportTopic,
-                  MegachusFullName,
-                  MegachusAddress
-                )
-              )
-              .flatten
-// Assert
-            megachusHouseInhabitants <- eventually(getHouseInhabitants(server.baseUri, MegachusAddress))
-            _ <- eventually(
-              assertIO(
-                getConsignment(server.baseUri, megachusHouseInhabitants.head),
-                ChristmasConsignment(
-                  MegachusFullName,
-                  MegachusAddress.some,
-                  ChristmasPresent.Nice(uri"nice_present", ExpectedScore)
-                )
-              )
-            )
-          } yield ()
-        }
-    }
+      )
+    } yield ()
+
   }
 
   private def getConsignment(
       baseUri: Uri,
       fullName: FullName
   ): IO[ChristmasConsignment] =
-    EmberClientBuilder.default[IO].build.use { client =>
-      client.expect[ChristmasConsignment](
-        baseUri.withPath(
-          Path.unsafeFromString(
-            s"/list/${fullName.lastName}/${fullName.firstName}"
-          )
+    fixture().httpClient.expect[ChristmasConsignment](
```

Comment:
so we do need to update the posts about this then :D

Reply:
There's permalinks to the exact revision in the posts (and we've copied the code into the posts) so there's no rush I guess.
But yeah, probably worthwhile.